docs/sections/calculations/errors.md

```diff
@@ -1,302 +1,61 @@
-
-Now, specify the number of machines (a.k.a. cluster nodes) you are going to run on and the maximum time allowed for the calculation.
-The general options grouped under `builder.metadata.options` are independent of the code or plugin, and will be passed to the scheduler that handles the queue on your compute resource.
+You will need to set the rest of the inputs by yourself, the most critical being the following:
 
 ```{code-block} ipython
 
-In [4]: builder.metadata.options.resources = {'num_machines': 1}
-   ...: builder.metadata.options.max_wallclock_seconds = 30 * 60
+In [2]: builder.structure = ...
+   ...: builder.kpoints = ...
+   ...: builder.pseudos = ...
+   ...: builder.metadata.options.resources = {'num_machines': 1}
 
 ```
 
-Again, to see the list of available options, type `builder.metadata.options.` and hit the TAB button.
-
-### Preparation of inputs
-
-A Quantum ESPRESSO calculation needs a number of inputs:
+For the `structure` you can download the following {download}`silicon crystal<include/data/Si.cif>` and import it into your database.
+If you have already loaded this structure in your database (as it is used in other tutorial modules), you may want to use that pre-existing node instead of saving a new node with repeated information.
+To do so you may search for its pk by running `verdi data structure list` and then use the python function `load_node()` to retrieve it into a variable.
 
-1. [Pseudopotentials](<https://en.wikipedia.org/wiki/Pseudopotential>)
-2. a structure
-3. a mesh in reciprocal space (k-points)
-4. a number of input parameters
+For the `pseudos` (or [pseudopotentials](<https://en.wikipedia.org/wiki/Pseudopotential>)), you can use the `SSSP/1.1/PBE/efficiency` family of the `aiida-pseudo` package.
+If you already have it installed, it is enough to use the `load_group` function and then the `get_pseudos` method of the pseudo group.
 
-These are mirrored in the inputs of the `aiida-quantumespresso` plugin (see [documentation](<https://aiida-quantumespresso.readthedocs.io/en/stable/user_guide/calculation_plugins/pw.html>)).
-We'll start with the structure, k-points, and pseudopotentials and leave the input parameters as the last thing to setup.
-
-:::{admonition} Exercise
-
-Use what you learned in the basics section to load the `structure` and `kpoints` inputs for your calculation:
-
-* Use a silicon crystal.
-* Define a `2x2x2` mesh of k-points.
-
-Note: If you just copy and paste code that you executed previously, this may result in duplication of information on your database.
-In fact, you can re-use an existing structure stored in your database [^f1].
-Use a combination of the bash command `verdi data structure list` and the python function `load_node()` to get an object representing the structure created earlier.
-
-:::
+The set of `kpoints` can be simply created by using the `KpointsData` plugin methods to define a `2x2x2` mesh and store it in a new node.
```

Comment:
Maybe there is some technicality that I'm missing here. I know that the `KpointsData` comes with AiiDA core, but is still considered a plugin in the general cathegorical sense right? I mean, it does appear when you run `verdi plugin list aiida.data` and it seems to be counted as "plugins provided by the package" in the [aiida-core plugin page](https://aiidateam.github.io/aiida-registry/plugins/core.html).

Reply:
If you understand it like this even `verdi` becomes a plugin...
No, that page is merely a list of all the various parts that `aiida-core` provides.
Furthermore, my point was that a plugin is not a single class, but rather a whole package. `aiida-quantumespresso` is a plugin, in the sense that it's a package that plugs into the `aiida-core` package through entrypoints, providing new calculation and data classes as well as parsers and more - maybe even another CLI.
So for this particular sentence to make sense I would always write something like "the `KpointsData` node class from the X plugin". In this case that doesn't even make sense since this comes from the core package, and the core package cannot by definition be a plugin to itself.

Reply:
> If you understand it like this even `verdi` becomes a plugin...
> No, that page is merely a list of all the various parts that `aiida-core` provides.

Mmm yeah that is true, I don't know why `verdi` appears listed like that in the page and it is a bit weird. I'm not 100% sure of the class structure of this, but I would expect the `verdi` command line to be some very different python thing than the `KpointsData` class. However he `KpointsData` is the same kind of python thing as, say, the `gaussian.basisset` or the `gaussian.pseudo` that comes with [aiida-gaussian-datatypes](https://aiidateam.github.io/aiida-registry/plugins/gaussian-datatypes.html) package: they are branches that connect to the same root of parent classes so to speak.

> Furthermore, my point was that a plugin is not a single class, but rather a whole package. `aiida-quantumespresso` is a plugin, in the sense that it's a package that plugs into the `aiida-core` package through entrypoints, providing new calculation and data classes as well as parsers and more - maybe even another CLI.

Ehm, I'm pretty sure this is not the case 🤔 . I mean, I may be misremembering, but I think last time we discussed this around a year ago we came to the decision of calling `aiida-quantumespresso` a **plugin package** and leave **plugin** for each of the components `quantumespresso.pw`, `quantumespresso.pp`, `quantumespresso.cp`, etc.

> So for this particular sentence to make sense I would always write something like "the KpointsData node class from the X plugin". In this case that doesn't even make sense since this comes from the core package, and the core package cannot by definition be a plugin to itself. At most it can be an extension to the more general Data node.

Ok, I have no problem changing the wording here since it is not so critical, but if you don't remember the same thing as I do above maybe we should check with the group to make sure what is the right convention for future reference.

Reply:
I think you might be right about the **plugin package** versus **plugin**. I'd probably personally refer to the latter as entry point plugins to the `aiida-core` package, but that still makes them **plugins**.

`KpointsData` is just a Python class. But since it's provided by the core package it can never be a plugin. Not in the normal sense anyway. However, if the AiiDA Team organization has re-defined the meaning of "plugin" with respect to AiiDA I might be wrong. I would still say any given Python class, function or variable can never be a plugin, but it is rather plugged in, i.e., it's the object of the plugging, not the plugging itself 😅